### ISSUE TITLE
”InvalidArgumentError” when implementing a GAN with batchnormalization

### ISSUE THREAD

**Issue by wkcw** [fchollet/keras#5752]
Hello,

I've encountered an error similar to [this issue](https://github.com/fchollet/keras/issues/5221)
At first, the error occurs when executing "d_loss = discriminator.train_on_batch(...)".
So I tried the solution in the link above (add discriminator._make_train_function() before building the combined model ).  And the "discriminator.train_on_batch()"  passed.

But when the script moves on to "g_loss = dual_dis0.train_on_batch(...)" (dual_dis0 is the combined gan model in my script). The similar error occurs again. I don't know how to solve it this time.

Because my script is a bit too long to put it here, may I put the [link of my code](https://github.com/wkcw/keras-face-attribute-manipulation) here?
the error can be reproduced with "python model$train.py" after cloning it.

**Comment by thepancake1** [comment]
I'm also having this exact issue. It seems that adding the Batch Normalization layer to a combined model causes this. 

`
InvalidArgumentError (see above for traceback): You must feed a value for placeholder tensor 'conv2d_1_input' with dtype float`

I'm really new to Keras, and machine learning in general. Why does this particular setup cause an issue?

**Comment by iNLyze** [comment]
Did you try a different setting for the mode parameter in BatchNorm? E.g. try mode=2

**Comment by iNLyze** [comment]
Note: In Keras2 mode parameter does not exist anymore. Make sure the axis parameter matches the backend framework, e.g. axis=-1 for tensorflow.
